# ==============================================================================
# REGRESSOR COMMIT DIFF
# ==============================================================================
# File: dom/canvas/WebGLTypes.h
# Commit: 16c5dd2cfd60
# Full Hash: 16c5dd2cfd600a62c80ca17cfb35c0868e016056
# Author: Jeff Gilbert <jgilbert@mozilla.com>
# Date: 2020-06-15 21:48:38
# Regressor Bug: 1632249
# File Overlap Count: 1
# Description:
#   Bug 1632249 - Support compositing out-of-process WebGL. r=handyman,lsalzman,nical,geckoview-reviewers,agi,imanol
#   
#   * Majorly simplity CanvasRenderer
#   * Replace GLScreenBuffer with trivial GLSwapChain
#   * Use descriptor structs so that future SharedSurface changes aren't so painful
# ==============================================================================

diff -r ca8139d58d58 -r 16c5dd2cfd60 dom/canvas/WebGLTypes.h
--- a/dom/canvas/WebGLTypes.h	Fri Jun 12 21:53:14 2020 +0000
+++ b/dom/canvas/WebGLTypes.h	Mon Jun 15 18:25:55 2020 +0000
@@ -989,6 +989,28 @@
   return {text.BeginReading(), text.Length()};
 }
 
+// -
+// ClientWebGLContext exports!
+// (So that you don't need to include the mess that is ClientWebGLContext.h)
+namespace gfx {
+class SourceSurface;
+}  // namespace gfx
+
+namespace layers {
+class KnowsCompositor;
+class SurfaceDescriptor;
+}  // namespace layers
+
+namespace webgl {
+
+void Present(ClientWebGLContext&);
+Maybe<layers::SurfaceDescriptor> GetFrontBuffer(ClientWebGLContext&,
+                                                layers::KnowsCompositor*);
+
+}  // namespace webgl
+
+// -
+
 }  // namespace mozilla
 
 #endif